GoodVibes.Client.Lovense/Models/Abstractions/LovenseToy.cs

```diff
@@ -153,34 +179,48 @@ private void AddCommandToList(LovenseCommandEnum function, int value)
         public string? GetCommandString()
         {
             if (ToyCommands.Count == 0) return null;
-            
-            var commandStr = string.Empty;
+
+            var function1Value = Function1LastValue;
+            var function2Value = Function2LastValue;
+            var function3Value = Function3LastValue;
+
             foreach (var toyCommand in ToyCommands)
             {
-                //var highestValue = 0;
-                var lastValue = 0;
-                if (toyCommand.Key == Function1 || toyCommand.Key == Function2)
-                {
-                    var values = toyCommand.Value;
-
-                    // It seems like this, if you're quick is actually null.
-                    if (values == null) continue;
+                var values = toyCommand.Value;
+                if (values == null) continue;
 
-                    // TODO: Add more calculation methods for different behaviors
-                    //highestValue = values.Prepend(0).Max();
-                    lastValue = values.Last();
+                if (toyCommand.Key == Function1)
+                {
+                    function1Value = values.Last();
                 }
-                else
+
+                if (toyCommand.Key == Function2)
                 {
-                    continue;
+                    function2Value = values.Last();
                 }
 
-                if (commandStr != string.Empty)
+                if (toyCommand.Key == Function3)
                 {
-                    commandStr += ",";
+                    function3Value = values.Last();
                 }
+            }
+
+            var commandStr = string.Empty;
+            if (Function2 == LovenseCommandEnum.None && Function3 == LovenseCommandEnum.None)
+            {
+                commandStr = $"{Function1.ToString()}:{DivideByStrengthPercentage(Function1, function1Value)}";
+            }
 
-                commandStr += $"{toyCommand.Key.ToString()}:{DivideByStrengthPercentage(toyCommand.Key, lastValue)}";
+            if (Function3 == LovenseCommandEnum.None)
+            {
+                commandStr =
+                    $"{Function1.ToString()}:{DivideByStrengthPercentage(Function1, function1Value)},{Function2.ToString()}:{DivideByStrengthPercentage(Function2, function2Value)}";
+            }
+
+            if (Function3 != LovenseCommandEnum.None)
+            {
+                commandStr =
+                    $"{Function1.ToString()}:{DivideByStrengthPercentage(Function1, function1Value)},{Function2.ToString()}:{DivideByStrengthPercentage(Function2, function2Value)},{Function3.ToString()}:{DivideByStrengthPercentage(Function3, function3Value)}";

```

Comment:
🤔 I feel like this could be optimised a bit and using an array
```js
var Functions = [ Function1, Function2, Function3 ]
var commandArr = [];
Functions.foreach(function(Function) {
   if (Function !== LovenseCommandEnum.None) {
      commandArr.push(
         $"{Function.ToString()}:{DivideByStrengthPercentage(Function, functionValue)},
      );
   }
});

var commandStr = commandArr.join(",");
```
This way if Lovense Add 50 commands this sort of thing is already ready for gluing the command together~

Reply:
I completely agree and I have now fixed it. No idea what I was thinking when I wrote that code tbh :)